test/integration/glue_test.py

```diff
@@ -1,121 +1,229 @@
-    unmapped_plants = db_plants.loc[new_plants_index].reset_index()
-    if len(unmapped_plants) != 1:
-        raise AssertionError(
-            f"Found {len(unmapped_plants)} unmapped FERC 1 plants instead of 1."
-        )
 
+@pytest.fixture(scope="module")
+def glue_test_dfs(
+    pudl_out, ferc1_engine_xbrl, ferc1_engine_dbf, pudl_settings_fixture
+) -> dict[str, pd.DataFrame]:
+    """Make a dictionary of the dataframes required for this test module."""
+    glue_test_dfs = {
+        "plants_eia_pudl_db": pudl_out.plants_eia860(),
+        "util_ids_ferc1_raw_xbrl": get_util_ids_ferc1_raw_xbrl(ferc1_engine_xbrl),
+        "util_ids_ferc1_raw_dbf": get_utils_ferc1_raw_dbf(ferc1_engine_dbf),
+        "plants_ferc1_raw": get_raw_plants_ferc1(
+            pudl_settings=pudl_settings_fixture,
+            years=DataSource.from_id("ferc1").working_partitions["years"],
+        ),
+    }
+    glue_test_dfs.update(glue(eia=True, ferc1=True))
+    return glue_test_dfs
 
-def test_unmapped_utils_ferc1(pudl_settings_fixture, ferc1_dbf_engine):
-    """Test that we can identify unmapped FERC 1 utilities."""
-    # First run the unmapped utility function as is:
-    actually_unmapped_utils = get_unmapped_utils_ferc1(ferc1_dbf_engine)
-    if not actually_unmapped_utils.empty:
-        raise AssertionError(
-            f"Found {len(actually_unmapped_utils)} unmapped FERC 1 utilities, "
-            f"expected 0.\n"
-            f"{actually_unmapped_utils}"
-        )
-    logger.info("Found 0 unmapped FERC 1 utilities, as expected.")
 
-    # Now do the smae thing... but yanking one of the mapped utils:
-    mapped_utilities = (
-        get_mapped_utils_ferc1()
-        .set_index("utility_id_ferc1")
-        .drop(145)  # Drop Xcel Energy Colorado / PSCo.
-    )
-    # Get all the plants in the FERC 1 DB:
-    db_plants = get_db_plants_ferc1(
-        pudl_settings_fixture, DataSource.from_id("ferc1").working_partitions["years"]
-    ).set_index(["utility_id_ferc1", "plant_name_ferc1"])
-    # Read in the mapped plants... but ditch Xcel's Comanche:
-    mapped_plants = (
-        get_mapped_plants_ferc1()
-        .set_index(["utility_id_ferc1", "plant_name_ferc1"])
-        .drop((145, "comanche"))
-    )
-    new_plants_index = db_plants.index.difference(mapped_plants.index)
-    unmapped_plants = db_plants.loc[new_plants_index].reset_index()
-    # Generate a list of all utilities which have unmapped plants:
-    # (Since any unmapped utility *must* have unmapped plants)
-    utils_with_unmapped_plants = (
-        unmapped_plants.loc[:, ["utility_id_ferc1", "utility_name_ferc1"]]
-        .drop_duplicates("utility_id_ferc1")
-        .set_index("utility_id_ferc1")
+def save_to_devtools_glue(df, test_dir, request):
+    """Save a dataframe as a CSV to the glue directory in devtools."""
+    file_path = Path(
+        test_dir.parent,
+        "devtools",
+        "ferc1-eia-glue",
+        f"{request.node.callspec.id}.csv",
```

Comment:
What magic is going on here with the filename? How are they output files named?

Reply:
Is there a good reason not to just pass in a string for use in the filename, rather than the whole request object, which isn't really being used here except for the string ID. In the normal case you'd get the string by passing in that ID, and you'd avoid having to modify the request object in the `missing_utils_eia` case.

Reply:
thats a good idea